docs/guides/developers.md

```diff
@@ -0,0 +1,49 @@
+# Developer guide
+This guide contains some more in-depth guidelines, tasks and examples that developers should adhere to when working on hydrolib-core.
+In addition to being a useful resource for developing, this documentation can also be helpful for code reviews.
+
+## Explicitly exposing the public API
```

Comment:
I think something went wrong with updating this part of the documentation

Reply:
Done